### ISSUE TITLE
Self Hosting: Fetching shared link does not work, Cannot GET /eval/<id>

### ISSUE THREAD

**Issue by AlexRRR** [promptfoo/promptfoo#1644]
**Describe the bug**
The shared link produced now has the structure:
```
 Evaluation complete: https://promptfoo.mycorp.net/eval/eval-2024-09-12T14:50:30
```
however trying always returns 400 `Cannot GET /eval/eval-2024-09-12T14:50:30` 

I am able to see the report in the UI at `https://promptfoo.mycorp.net/eval/?evalId=eval-2024-09-12T14:50:30`

**To Reproduce**
Self hosting
- Build 0.86.1 docker image 
- run it with set envvar 
```
PROMPTFOO_REMOTE_API_BASE_URL=https:///promptfoo.mycorp.net
```

**System information (please complete the following information):**

- Promptfoo version: 0.86.1

I can clearly see there is a huge pace in versions, but I am struggling to find a combination that works for sharing the reports. Do you have any pointers how this is done?

**Comment by typpo** [comment]
Hi @AlexRRR, can you try the latest Dockerfile on main?  That should solve the issue, which is a UI bug in how the share ID is presented.

**Comment by AlexRRR** [comment]
Thanks for the hint @typpo . However I am hitting the same problem, I've built dd0ad14 and still i get a `400` with `Cannot GET /eval/eval-2024-09-13T08:00:08` but I am able to see it under `/eval/?evalId=eval-2024-09-13T08%3A00%3A08` 

To be clear I've only set `PROMPTFOO_REMOTE_API_BASE_URL:  https://promptfoo.mycorp.net` as envvar, perhaps I am missing something else.

Also I found a small bug while building this commit's docker file:
```
26.60 **************************************************
26.60 Building next.js
26.60 **************************************************
26.78 Attention: Next.js now collects completely anonymous telemetry regarding usage.
26.78 This information is used to shape Next.js' roadmap and prioritize features.
26.78 You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
26.78 https://nextjs.org/telemetry
26.78 
26.92 - info Creating an optimized production build...
63.71 thread '<unnamed>' panicked at 'internal error: entered unreachable code', /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rayon-core-1.11.0/src/job.rs:230:32
63.71 note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
63.73 fatal runtime error: failed to initiate panic, error 5
63.78 Aborted
64.07 cp: no such file or directory: src/web/nextui/out/*
```
I've just done `touch src/web/nextui/out/foo`  and then dockerfile worked again.

**Comment by mldangelo** [comment]
Hi @AlexRRR, we’ve just shipped [0.88.0](https://github.com/promptfoo/promptfoo/releases/tag/0.88.0), which includes a fix for the shared link issue you reported and the missing directory. It should now be resolved. Thanks again for bringing this to our attention, and let us know if you run into any further issues!

**Comment by AlexRRR** [comment]
Thank you so much!
